src/glossary/index.md

```diff
@@ -0,0 +1,454 @@
+
+When a function is described as having side effects, it means that the function performs some sort of action that is observable outside the function, aside from just returning a value. This might mean that it updates a value in state, or triggers a network request.
+
+The term is often used when describing rendering or computed properties. It is considered best practice for rendering to have no side effects. Likewise, the getter function for a computed property should have no side effects.
+
+## Single-File Component {#single-file-component}
+
+The term *Single-File Component*, or SFC, refers to the `.vue` file format that is commonly used for Vue components.
+
+See also:
+- [Guide - Single-File Components](/guide/scaling-up/sfc.html)
+- [SFC Syntax Specification](/api/sfc-spec.html)
+
+## slot {#slot}
+
+Slots are used to pass content to child components. Whereas props are used to pass data values, slots are used to pass richer content consisting of HTML elements and other Vue components.
+
+For more details see:
+- [Guide - Slots](/guide/components/slots.html)
+
+## template ref {#template-ref}
+
+The term *template ref* refers to using a `ref` attribute on a tag within a template. After the component renders, this attribute is used to populate a corresponding property with either the HTML element or the component instance that corresponds to the tag in the template.
+
+If you are using the Options API then the refs are exposed via properties of the `$refs` object.
+
+With the Composition API, template refs populate a reactive [ref](#ref) with the same name.
+
+Template refs should not be confused with the reactive refs found in Vue's reactivity system.
+
+For more details see:
+- [Guide - Template Refs](/guide/essentials/template-refs.html)
+
+## VDOM {#vdom}
+
+See [virtual DOM](#virtual-dom).
+
+## virtual DOM {#virtual-dom}
+
+The term *virtual DOM* (VDOM) is not unique to Vue. It is a common approach used by several web frameworks for managing updates to the UI.
+
+Browsers use a tree of nodes to represent the current state of the page. That tree, and the JavaScript APIs used to interact with it, are referred to as the *document object model*, or *DOM*.
+
+Manipulating the DOM is a major performance bottleneck. The virtual DOM provides one strategy for managing that.
+
+Rather than creating DOM nodes directly, Vue components generate a description of what DOM nodes they would like. These descriptors are plain JavaScript objects, known as VNodes (virtual DOM nodes). Creating VNodes is relatively cheap.
+
+Every time a component re-renders, the new tree of VNodes is compared to the previous tree of VNodes and any differences are then applied to the real DOM. If nothing has changed then the DOM doesn't need to be touched.
+
```

Comment:
Worth adding a note that Vue improves virtual DOM runtime performance with compilers, since many new users nowadays assume virtual DOM to be associated with bad performance due to Svelte / Solid's marketing.